compiler-cli/integration-tests/semantic/if_else_always_returns.mj

```diff
@@ -0,0 +1,18 @@
+class Foo {
+
+      public int foo() {
+         if (true) {
+           return 5;
+         }
+
+         return 42;
+      }
+
+      public int bar() {
+         while (true) {
+           return 5;
+         }
+      }
+
+      public static void main(String[] args) {}
```

Comment:
Btw. Shouldn't there be a test where the void returning method is not the special case "main method"

Reply:
Hm? Every method can have the return type `void`